hack/make-and-check-size

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+#
+# make-and-check-size - wrapper around 'make' that also checks binary growth
+#
+# This script is intended to be run via 'git rebase -x', in a Makefile rule
+# such as:
+#
+#    build-all-new-commits:
+#       CONTEXT_DIR=$(shell mktemp -d --tmpdir make-size-check.XXXXXXX); \
+#       git rebase $(GIT_BASE_BRANCH)^ -x "hack/make-and-check-size $$CONTEXT_DIR"; \
+#       $(RM) -rf $$CONTEXT_DIR
+#
+# ...which has long been a part of our usual CI, one that makes sure that
+# each commit (in a multi-commit PR) can be compiled individually. By
+# adding the '^' to GIT_BASE_BRANCH we establish a baseline and store
+# the binary sizes of each file (podman, podman-remote) prior to our PR.
+#
+# CONTEXT_DIR is a temporary directory used to store the original sizes
+# of each binary file under bin/
+#
+# *IMPORTANT NOTE*: this script will leave the git checkout in a funky state!
+# (because we rebase onto a nonterminal commit). I believe this is OK, since
+# this makefile target is used only in CI and only in a scratch VM. Running
+# this in a development environment would yield unpredictable results anyway,
+# by rebasing onto origin/main by default and by leaving an aborted rebase
+# on failure.
+#
+ME=$(basename $0)
+
+# Maximum allowable size, in bytes. Can be overridden via github description
+MAX_BIN_GROWTH=$((100 * 1024))
+override=$(expr "${CIRRUS_CHANGE_MESSAGE}" : ".*MAX_BIN_GROWTH=\([0-9]\+\)")
+if [[ -n "$override" ]]; then
+    echo "*** Overriding default MAX_BIN_GROWTH($MAX_BIN_GROWTH) with $override"
+    MAX_BIN_GROWTH="$override"
+fi
+
+set -e
+
+# Must be invoked with one argument, an existing context directory
+context_dir=${1?Missing CONTEXT-DIR argument}
+if [[ ! -d $context_dir ]]; then
+    echo "$ME: directory '$context_dir' does not exist"
+    exit 1
+fi
+
+make
+
+# Determine size of each built file.
+# - If this is our first time through, preserve that size in a tmpfile
+# - On all subsequent runs, compare built size to initial size
+for bin in bin/*;do
+    size=$(stat -c %s $bin)
+
+    saved_size_file=$context_dir/$(basename $bin)
+    if [[ -e $saved_size_file ]]; then
+        # Not the first time through: compare to original size
+        size_orig=$(< $saved_size_file)
+        delta_size=$(( size - size_orig ))
+
+        if [[ $delta_size -gt $MAX_BIN_GROWTH ]]; then
+            cat <<EOF >&2
+
+*******************************************************************************
+* $bin grew by $delta_size bytes; max allowed is $MAX_BIN_GROWTH.
+*
+* Please investigate. You can override by editing your Github PR description
+* and adding:
+*
+*   [MAX_BIN_GROWTH=xxxxx]    (where xxxxx > $MAX_BIN_GROWTH)
```

Comment:
This was something I really worried about because somewhat like the tests, it will get abused.  The goal of this action would be that maintainers are notified; whereas once this override is known, there will be no forced method of maintainers seeing it.  I was hoping the task would fail ... and then we would use a github LABEL to override. I have seen this in other repos.

Reply:
That was my concern too (abuse). I can't find any sane way to get PR labels from within Cirrus, and am totally stumped trying to think of other possibilities. [Posting idea-free comment anyway because I've been stuck on this for 30m and y'all are discussing it on IRC separately]